### PR TITLE
Marcel/bees

### PR DESCRIPTION
**Pull request by fatboypunk**

### CODE REVIEW

File: apps/game/lib/bee.ex [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  @enforce_keys [:pos_x, :pos_y]
+  defstruct [:id, :catching, :pos_x, :pos_y]
+
+  def start_link(defaults) do
+    GenServer.start_link(__MODULE__, defaults)
+  end
+
+  def init(position) do
+    :timer.send_interval(100, self(), :update)
+    GenServer.cast(Game, {:create_bee, self})
```
Comment (Blackjaguar18) [inline apps/game/lib/bee.ex]: could use `self()` here, for clarity

File: apps/game/lib/bee.ex [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    {:ok, pid}
```
Comment (Blackjaguar18) [inline apps/game/lib/bee.ex]: where does this pid come from? why do we need the pid as state

File: apps/game/lib/game_server.ex [hunk 3]
```diff
@@ -363,6 +418,15 @@ defmodule Game do
     GenServer.call(Game, :get_players)
   end
 
+  defp nest_to_bee?(%Bee{pos_x: pox_x, pos_y: pos_y}, bear)
```
Comment (Blackjaguar18) [inline apps/game/lib/game_server.ex]: `next_to_bee?`

File: apps/game/lib/bee.ex [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+  end
+
+  @impl true
+  def handle_info(:update, pid) do
+    bee = GenServer.call(Game, {:get_bee, pid})
```
Comment (Blackjaguar18) [inline apps/game/lib/bee.ex]: cant we pass the bee around as state?

File: apps/game/lib/game_server.ex [hunk 5]
```diff
@@ -205,6 +243,17 @@ defmodule Game do
       Enum.find(trees, &(&1.pos_x == target_x and &1.pos_y == target_y))
   end
 
+  def update_state_with(%{bees: bees} = state, bee = %Bee{}) do
+    bees =
+      Enum.map(bees, fn list_bee ->
+        if list_bear.id == bee.id,
```
Comment (Blackjaguar18) [inline apps/game/lib/game_server.ex]: `list_bee`, same a few lines down

File: apps/game/lib/bee.ex (hunk 4 above)
Comment (Blackjaguar18) [reply apps/game/lib/bee.ex]: Nevermind, not possible

File: apps/game/lib/bee.ex [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+  @enforce_keys [:pos_x, :pos_y]
+  defstruct [:id, :catching, :pos_x, :pos_y]
+
+  def start_link(defaults) do
+    GenServer.start_link(__MODULE__, defaults)
+  end
+
+  def init(position) do
+    :timer.send_interval(100, self(), :update)
+    GenServer.cast(Game, {:create_bee, self})
```
Comment (Blackjaguar18) [inline apps/game/lib/bee.ex]: this cast needs an implementation

**Comment by fatboypunk** [comment]
Think everything is fixed now @Blackjaguar18 bees now also spawn at random when clawing a tree.

I think we could merge this when we have a better bee/background combination

File: apps/game/lib/bee.ex [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+  end
+
+  def terminate(reason, state) do
+    GenServer.cast(Game, {:remove_bee, self()})
+    reason
+  end
+
+  @impl true
+  def handle_info(:update, duration) when duration > 0 do
+    with %Bear{} = bee = GenServer.call(Game, {:get_bee, self}),
```
Comment (EasterPeanut) [inline apps/game/lib/bee.ex]: %Bee{}*

**Pull request merged**